src/net/sourceforge/kolmafia/KoLCharacter.java

```diff
@@ -4090,54 +4067,27 @@ public static final boolean hasAmphibianSympathy() {
     return KoLCharacter.hasSkill("Amphibian Sympathy");
   }
 
-  /** Utility method which looks up whether or not the character has a skill of the given name. */
-  public static final boolean hasSkill(final int skillId) {
-    UseSkillRequest skill = UseSkillRequest.getUnmodifiedInstance(skillId);
-    return KoLCharacter.hasSkill(skill);
-  }
-
-  public static final boolean hasSkill(final String skillName) {
-    // *** Skills can have ambiguous names. Best to use the methods that deal with skill id
-    UseSkillRequest skill = UseSkillRequest.getUnmodifiedInstance(skillName);
-    return KoLCharacter.hasSkill(skill);
+  /** Utility methods which looks up whether or not the character has a particular skill. */
+  public static final boolean hasAvailableSkill(final int skillId) {
+    return KoLConstants.availableSkillsSet.contains(skillId);
   }
 
-  public static final boolean hasSkill(final String skillName, final List<UseSkillRequest> list) {
-    // *** Skills can have ambiguous names. Best to use the methods that deal with skill id
-    UseSkillRequest skill = UseSkillRequest.getUnmodifiedInstance(skillName);
-    return KoLCharacter.hasSkill(skill, list);
+  public static final boolean hasCombatSkill(final int skillId) {
+    return KoLConstants.availableCombatSkillsSet.contains(skillId);
   }
 
-  public static final boolean hasSkill(final UseSkillRequest skill) {
-    return KoLCharacter.hasSkill(skill, KoLConstants.availableSkills);
+  public static final boolean hasSkill(final int skillId) {
+    return hasAvailableSkill(skillId);
   }
 
-  public static final boolean hasSkill(
-      final UseSkillRequest skill, final List<UseSkillRequest> list) {
-    if (skill == null) {
-      return false;
-    }
-
-    if (list == KoLConstants.availableSkills) {
-      return KoLConstants.availableSkillsMap.containsKey(skill);
-    }
-    if (list == KoLConstants.availableCombatSkills) {
-      return KoLConstants.availableCombatSkillsMap.containsKey(skill);
-    }
-    return list.contains(skill);
+  public static final boolean hasSkill(final String skillName) {
+    // *** Skills can have ambiguous names. Best to use the methods that deal with skill id
+    int skillId = SkillDatabase.getSkillId(skillName);
+    return KoLCharacter.hasSkill(skillId);
   }
 
   public static final boolean availableCombatSkill(final int skillId) {
-    UseSkillRequest skill = UseSkillRequest.getUnmodifiedInstance(skillId);
-    return KoLCharacter.availableCombatSkill(skill);
-  }
-
-  public static final boolean availableCombatSkill(final UseSkillRequest skill) {
-    if (skill == null) {
-      return false;
-    }
-
-    return KoLConstants.availableCombatSkillsMap.containsKey(skill);
+    return KoLConstants.availableCombatSkillsSet.contains(skillId);
```

Comment:
`availableCombatSkill` is identical to `hasCombatSkill`. It would be nice to only have one.

Similarly, `hasSkill` just calls `hasAvailableSkill`.

I like the symmetry in the names `hasSkill` and `hasCombatSkill`.

Reply:
Done.